src/components/forms/signInForm.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect } from 'react';
+import { FieldValues, useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { ThunkDispatch } from '@reduxjs/toolkit';
+import isAuthenticatedProperly from '../../utils/authHelper';
+import { fetchTokenRequest, refreshTokenRequest } from '../../redux/actions/token/tokenActions';
+import { FetchTokenRequestPayload, RefreshTokenRequestPayload } from '../../redux/types/type';
+import SubmitButton from '../../shared/buttons/SubmitButton';
+import { ITokenInput } from '../../models/token/ITokenInput';
+import Input from '../../shared/inputs/input';
+import { sessionToken } from '../../redux/reducers/tokenReducer';
+import { Paths } from '../../routes/paths';
+
+interface ISignInFormProps {
+  signIn: (params: FetchTokenRequestPayload) => void;
+  refresh: (params: RefreshTokenRequestPayload) => void;
+}
+
+type AuthAction = {
+  type: string,
+  err?: unknown
+};
+
+function SignInForm({ refresh, signIn }: ISignInFormProps) {
+  const navigate = useNavigate();
+
+  const callback = () => {
+    navigate(Paths.landing);
+  };
+
+  const refreshToken = () => {
+    const token = JSON.parse(localStorage.getItem(sessionToken) as string);
+    const data: any = {
+      values: {
+        refresh: token.refresh,
+      },
+      callback,
+    };
+    refresh(data);
+  };
+
+  useEffect(() => {
+    const isToken = !!localStorage.getItem(sessionToken);
+    if (isAuthenticatedProperly()) {
+      navigate(Paths.landing);
+    } else if (isToken) {
+      refreshToken();
+    }
+  });
+
+  const login = (formValues: ITokenInput) => {
+    const data: any = {
+      values: formValues,
+      callback,
+    };
+    signIn(data);
+  };
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = (formValues: FieldValues) => {
+    login(formValues as ITokenInput);
+  };
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} className="rounded-lg px-8 pt-6 pb-8 mb-4 space-y-4 ">
```

Comment:
From here remove className and add `<div className="flex flex-col gap-6">` - like that will be easier to control space between child div using in one place `gap` https://tailwindcss.com/docs/gap